packages/@coorpacademy-components/src/molecule/banner/index.js

```diff
@@ -1,39 +1,86 @@
-import React from 'react';
+import React, {useState, useCallback} from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
-import {keys} from 'lodash/fp';
-import {
-  NovaSolidInterfaceFeedbackInterfaceAlertDiamond as QuestionIcon,
-  NovaCompositionCoorpacademyValidate as ValidateIcon
-} from '@coorpacademy/nova-icons';
+import {keys, noop, isEmpty, map} from 'lodash/fp';
+import {COLORS} from '../../variables/colors';
 import ButtonLink from '../../atom/button-link';
+import Icon from '../../atom/icon';
+import InputSwitch from '../../atom/input-switch';
 import style from './style.css';
 
-const ICONS = {
-  success: ValidateIcon,
-  error: QuestionIcon,
-  warning: QuestionIcon
+const uncappedMap = map.convert({cap: false});
+
+const TYPES = {
+  success: ['circle-check', COLORS.cm_positive_200],
+  error: ['triangle-exclamation', COLORS.cm_negative_200],
+  warning: ['circle-exclamation', COLORS.cm_yellow_400],
+  info: ['circle-info', COLORS.cm_grey_500]
 };
 
 const STYLES = {
   success: style.success,
   error: style.error,
-  warning: style.warning
+  warning: style.warning,
+  info: style.info
 };
 
 const Banner = props => {
-  const {
-    type,
-    message,
-    firstCTA,
-    firstCTALabel,
-    secondCTALabel,
-    secondCTA,
-    temporary,
-    bannerKey,
-    onEnd
-  } = props;
-  const Icon = ICONS[type];
+  const {type, message, cta = [], temporary, bannerKey, onEnd} = props;
+  const [switchValue, setSwitchValue] = useState(false);
+
+  const handleSwitchToggle = useCallback(
+    action => () => {
+      setSwitchValue(!switchValue);
+      action();
+    },
+    [switchValue, setSwitchValue]
+  );
+
+  const ButtonSeparator = <div className={classnames(style.buttonsBar, STYLES[type])} />;
+
+  const buildButton = ({type: buttonType = 'button', label, action = noop}) => {
+    switch (buttonType) {
+      case 'switch':
+        return (
+          <div className={classnames(style.button, STYLES[type])}>
+            <InputSwitch
+              theme="coorpmanager"
+              title={label}
+              value={switchValue}
+              onChange={handleSwitchToggle(action)}
+              data-name={`banner-switch-cta`}
+              aria-label={label}
+            />
+          </div>
+        );
+      default:
+        return (
+          <div className={classnames(style.button, STYLES[type])}>
+            <ButtonLink
+              data-name="banner-button-cta"
+              aria-label={label}
+              label={label}
+              onClick={action}
+              type="text"
+            />
+          </div>
+        );
+    }
+  };
+
+  const buildCta = (options, i) => {
```

Comment:
tu peux rnetrer le map dedans, c'est weird la.

Reply:
if you want but why it's weird? 😅